File: src/main/java/com/mbayou/kick4k/events/type/LivestreamMetadata.java
package com.mbayou.kick4k.events.type;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.mbayou.kick4k.categories.Category;

public class LivestreamMetadata {
    private final String title;
    private final String language;
    private final Boolean hasMatureContent;
    private final Category category;

    public LivestreamMetadata(@JsonProperty("title") String title,
                              @JsonProperty("language") String language,
                              @JsonProperty("has_mature_content") Boolean hasMatureContent,
                              @JsonProperty("category") Category category) {
        this.title = title;
        this.language = language;
        this.hasMatureContent = hasMatureContent;
        this.category = category;
    }

    public String getTitle() {
        return title;
    }

    public String getLanguage() {
        return language;
    }

    public Boolean getHasMatureContent() {
        return hasMatureContent;
    }

    public Category getCategory() {
        return category;
    }
}

File: src/main/kotlin/com/mbayou/kick4k/events/type/LivestreamMetadata.kt
package com.mbayou.kick4k.events.type

import com.fasterxml.jackson.annotation.JsonAlias
import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.mbayou.kick4k.categories.Category

data class LivestreamMetadata @JsonCreator constructor(
    @JsonProperty("title") val title: String,
    @JsonProperty("language") val language: String,
    @JsonProperty("has_mature_content") val hasMatureContent: Boolean,
    @JsonProperty("category") @JsonAlias("Category") val category: Category,
)
